### ISSUE TITLE
Pipe tables / HTML: option to disable the column width computation algorithm for long lines

### ISSUE THREAD

**Issue by sphakka** [jgm/pandoc#10111]
It is my understanding that, with HTML output, the rendering of (pipe) tables depends on (at least) two things:

1. The column width (option `--columns`), and
2. The structure of the header separator line which may enforce wrapping and/or different widths of the columns.

Now, I find it a bit difficult to foresee the results and, IMHO, it looks like a violation of POLA. My point is that there can be many scenarios where one would expect a table to be rendered with the simplest minimal HTML code -- I make heavy use of CSS for styling my documents.

In my case, editing with Emacs's markdown mode is problematic: indeed, this mode has a handy table-alignment function that expands all fields to give a normalized view -- it doesn't add any MD/HTML formatting though. But of course, as a recurring result, lines are often longer than the default value for `--columns`, which screws the rendering with extra inline style and colgroups.

Workarounds like `--columns=1000` (mmh, possible unexpected side effects) or CSS rules `width: auto !important;` (ugly) can be used, though a better long term solution would be an option to entirely disable the algorithm. Something like `--smart-tables`, possibly `false` by default.

**Comment by jgm** [comment]
I take your point. Unfortunately many formats do need the relative width information, as they can't intelligently compute this based on the content.  HTML is an exception.  Perhaps we could add an extension, say `smart_tables`, and have it default to true.

**Comment by terefang** [comment]
this is exactly the opposite of my use-case.

could it be that we can have both an option like `--compute-table-widths=on|off|auto`

with:

* "auto" being the behavior as it is now
* "off" never compute them as for @sphakka case
* "on" always compute them as for my case.

sorry that i highjack this issue.

**Comment by tarleb** [comment]
My interpretation of the above is that the new extension `smart_tables`, should it be introduced, would effectively cause all columns to have `ColWidthDefault` *when the extension is disabled*. Is that correct?

**Comment by jidanni** [comment]
All I know is HTML tables render fine without any "help" from this stuff which I had to filter out
```
perl -pwle 's!style="width:\s*\d+%;*"!!'
```
so please add an option to turn it off. Thanks.

**Comment by terefang** [comment]
@jidanni problem is today col-widths are calculated only if the table width is greater than X,, with X begin the number of text-columns set (eg. `--columns=...`).

i want to have columns widths to always calculated (ie the future +smart_tables) regardless of columns setting since a too low setting can have other side-effects..

you can get what you want by setting `--columns=2000`.

**Comment by jidanni** [comment]
Time for a `--no-artificially-added-widths-please` option, else hack options are just too confusing.

**Comment by terefang** [comment]
@jidanni  yes i know that the default behavior choice was unfortunate, but i also understand the developers stance on not break existing behavior if it isnt buggy.

but the `--smart-table-widths=off` in concert with the proposed `smart_tables` extension would be exactly your (and @sphakka s) use-case if i am not mistaken.

**Comment by tarleb** [comment]
A Lua filter can be used to simulate the `smart_tables` extension:

``` lua
function Table (tbl)
  -- A colspec is a tuple, where the first element is the column's cell
  -- alignment, and the second element is the column width. Omitting that
  -- value causes pandoc to use the default / automatic width.
  tbl.colspecs = tbl.colspecs:map(function (spec) return {spec[1]} end)
  return tbl
end

```

I'd recommend to put this into a file `$PANDOC_DATA_DIR/filters/smart-tables.lua`, where `PANDOC_DATA_DIR` is the directory is the one listed by `pandoc -v`. The filter can then be used by running `pandoc -L smart-tables.lua …`.
